storage/engine/mvcc.go

```diff
@@ -939,14 +947,23 @@ func MVCCIterate(engine Engine, key, endKey proto.Key, timestamp proto.Timestamp
 		return key, iter.ValueProto(msg)
 	}
 
+	// A cumulative write intent error to gather all write intents.
+	var wiErr error
```

Comment:
A nil interface value is different from a typed nil pointer: http://golang.org/doc/faq#nil_error

http://play.golang.org/p/UVlgMpAxHL